src/framegrab/grabber.py

```diff
@@ -124,6 +125,35 @@ def create_grabbers(configs: List[dict], warmup_delay: float = 1.0) -> dict:
 
         return grabbers
 
+    @staticmethod
+    def from_yaml(filename: Optional[str] = None, yaml_str: Optional[str] = None) -> List["FrameGrabber"]:
+        """Creates multiple FrameGrab objects based on a YAML file or YAML string.
+        Either filename or yaml_str must be provided, but not both.
+
+        :param filename: The filename of the YAML file to load.
+        :param yaml_str: A YAML string to parse.
+        :return: A dictionary where the keys are the camera names, and the values are FrameGrabber objects.
+        """
+        if filename is None and yaml_str is None:
+            raise ValueError("Either filename or yaml_str must be provided.")
+        if filename is not None and yaml_str is not None:
+            raise ValueError("Only one of filename or yaml_str can be provided.")
+        if filename:
+            with open(filename, "r") as f:
+                yaml_str = f.read()
+        full_config = yaml.safe_load(yaml_str)
+        if "image_sources" not in full_config:
+            raise ValueError("Invalid config file. Camera configs must be under the 'image_sources' key.")
+        image_sources = full_config["image_sources"]
+        # Check that it's a list.
+        if image_sources is None or not isinstance(image_sources, list):
+            raise ValueError("Invalid config file. 'image_sources' must be a list.")
+        grabber_dict = FrameGrabber.create_grabbers(image_sources)
+        grabber_list = []
+        for _, grabber in grabber_dict.items():
```

Comment:
Is this expressing a desire to have create_grabbers return a list instead of a dict? The return type is something I went back and forth on. I ended up choosing a dict because it seemed convenient in the URCap to specify a grabber by name, but I can also see why in other use cases it might be simpler to have a list.

Reply:
I see both as useful.  It's probably better to be consistent.  For the very simple case I'm writing now, I just want to use `[0]`.